shtdlib.sh

```diff
@@ -2227,6 +2223,54 @@ function test_shopt_decorator {
     assert shopt -qo pipefail && color_echo green "Successfully decorated  ${FUNCNAME[0]} with pipefail"
 }
 
+# Test signaling
+function test_signal_process {
+    signal_processor SIGUSR2 'exit 42' > /dev/null
+    local sub_pid_0="${!}"
+    signal_processor SIGUSR1 "sleep 2 && kill -s SIGUSR2 ${sub_pid_0} && exit 42" > /dev/null
+    local sub_pid_1="${!}"
+    debug 10 "Spawned sub processes using signal processor with pids: ${sub_pid_0} and ${sub_pid_1}"
+    debug 10 "Active sub processes are: $(pgrep -P ${$} | tr '\n' ' ')"
+    signal_process "${sub_pid_1}" SIGUSR1 > /dev/null
+    debug 10 "Waiting for sub processes to exit"
+    bash -c "sleep 10 && kill ${sub_pid_0} &> /dev/null" &
+    bash -c "sleep 10 && kill ${sub_pid_1} &> /dev/null" &
+    while pgrep -P ${$} > /dev/null ; do
+        wait ${sub_pid_0} &> /dev/null
+        # Make sure the sub process exits with 42
+        assert [ "${?}" == 42 ]
+        color_echo green "Sub process was signaled, responded and properly exited"
+        return 0
+    done
+    color_echo red "Signaling and sub process test failed"
+    return 1
+}
+
+# Test filesystem monitoring/event triggers
+function test_add_on_mod {
+    signal_processor SIGUSR1 'exit 42' > /dev/null
+    local signaler_pid="${!}"
```

Comment:
Same as before.. not sure if this is necessary if `signal_processor` is echoing the same pid

Reply:
same as above, in case of failure...